wordpress-mobile/WordPress-iOS
# Issues/4094 reader menu restore and sync

Refs #4094
This PR implements state restoration for the Reader. 

To test:
Test state restoration via the following paths. Confirm you are always returned to the VC you were last on the next time you launch the app. 
- [x] Menu > Tag List > Post Detail
- [x] Menu > Following > Preview Tag > Preview Tag > Post Detail
- [x] Menu > Search > Preview Site > Post Detail
- [x] Menu > Likes > Comments
- [x] Menu > Manage > Preview Site > Post Detail > Comments



@jleandroperez  would you be up for the review? 
Needs review: @jleandroperez
